datamol/viz/_lasso_highlight.py

```diff
@@ -0,0 +1,439 @@
+            pos_list1 = _arch_points(r, points[0], points[1], 20)
+            # Translating arch from origin to atom position
+            pos_list1[:, 0] += at_manager.pos[0]
+            pos_list1[:, 1] += at_manager.pos[1]
+            # Transforming points to RDKit Objects
+            points = [Point2D(*c) for c in pos_list1]
+            canvas.DrawPolygon(points)
+
+        # Drawing lines parallel to each bond
+        for bond in at_manager.bonds:
+            if bond.bond_id in added_bonds:
+                continue
+            added_bonds.add(bond.bond_id)
+            bnd_points = at_manager.bond_attachment_points[bond.bond_id]
+
+            atom_i_left_at = _angle_to_coord(at_manager.pos, *bnd_points[0])
+            atom_i_right_at = _angle_to_coord(at_manager.pos, *bnd_points[1])
+            atom_j = a_obj_dict[bond.neighbour_id]
+            atom_j_left_at = _angle_to_coord(
+                atom_j.pos, *atom_j.bond_attachment_points[bond.bond_id][0]
+            )
+            atom_j_right_at = _angle_to_coord(
+                atom_j.pos, *atom_j.bond_attachment_points[bond.bond_id][1]
+            )
+            canvas.DrawLine(Point2D(*atom_i_left_at), Point2D(*atom_j_right_at))
+            canvas.DrawLine(Point2D(*atom_i_right_at), Point2D(*atom_j_left_at))
+    # restoring prior line width
+    canvas.SetLineWidth(prior_lw)
+
+
+# TODO switch this over to other doc string
+def _draw_multi_matches(
+    canvas: rdMolDraw2D.MolDraw2D,
+    mol: Chem.Mol,
+    indices_set_lists: List[Union[list, str]],
+    r_min: float = 0.3,
+    r_dist: float = 0.13,
+    relative_bond_width: float = 0.5,
+    color_list: List[ColorTuple] = None,
+    line_width: int = 2,
+) -> None:
+    """
+    Parameters
+    ----------
+    canvas : rdMolDraw2D.MolDraw2D
+        RDKit Canvas, where highlighting is drawn to.
+    mol: Chem.Mol
+        Atoms from the molecule `mol` are takes as positional reference for the highlighting.
+    indices_set_lists: List[Union[list, str]]
+        Atom indices delineating highlighted substructure.
+    r_min: float
+        Radius of the smallest circle around atoms. Length is relative to average bond length (1 = avg. bond len).
+    r_dist: float
+        Incremental increase of radius for the next substructure.
+    relative_bond_width: float
+        Distance of line to "bond" (line segment between the two atoms). Size is relative to `atom_radius`.
+    line_width: int
+        width of drawn lines.
+    color_list: List[ColorTuple]
+        List of tuples with RGBA or RGB values specifying the color of the highlighting.
+    Returns
+    -------
+    None
+    Returns
+    -------
+    """
+    # If no colors are given, all substructures are depicted in gray.
+    if color_list is None:
+        color_list = [(0.5, 0.5, 0.5)] * len(indices_set_lists)
+    if len(color_list) < len(indices_set_lists):
+        raise ValueError("Not enough colors for referenced substructures!")
```

Comment:
Instead of throwing an error, it seems to be best to simply iterate to +inf no?

```
color_list = [col1, col2, col3]

if <color_list is too small>:
   color_list = [col1, col2, col3] * XX
```